feat: 출근/퇴근 관리 API 구현

junbem opened the pull request:
- 출근 체크인/체크아웃 API 추가
- 근무 상태 관리 API 추가 (휴가, 출장, 재택 등)
- 주간 근무 요약 조회 API 추가
- PostgreSQL 데이터베이스 연동
- JPA 엔티티 및 리포지토리 구현
- 하드코딩된 DB 접속 정보를 환경 변수로 변경
- 기본값 설정으로 로컬 개발 환경 지원
- 민감한 정보의 GitHub 노출 방지

junbem committed:
feat: 출근/퇴근 관리 API 구현

- 출근 체크인/체크아웃 API 추가
- 근무 상태 관리 API 추가 (휴가, 출장, 재택 등)
- 주간 근무 요약 조회 API 추가
- PostgreSQL 데이터베이스 연동
- JPA 엔티티 및 리포지토리 구현

junbem committed:
feat: API 응답 형식 표준화

- common 패키지에 ApiResponse<T> 클래스 추가
- AttendanceController에서 일관된 응답 형식 반환하도록 수정
- 모든 엔드포인트에 에러 처리 및 의미있는 메시지 추가
- status, message, data 필드를 포함한 API 응답 구조 개선

junbem committed:
Merge branch 'develop' into feature/attendanceservice

junbem committed:
refactor: attendance-service DTO 구조 개선 및 API 표준화

- DTO 중복 제거: WeeklyWorkBase 상속 구조 도입
- API 응답 표준화: ApiResponse<T> 래퍼 적용
- 주간 조회 API 분리: 상세/통계 정보 별도 엔드포인트
- @SuperBuilder 사용으로 상속 구조에서 Builder 패턴 지원

hyeblee commented on attendance-service/src/main/java/com/hermes/attendanceservice/controller/AttendanceController.java:
```diff
@@ -0,0 +1,165 @@
+    private final AttendanceService attendanceService;
+
+    @PostMapping("/check-in")
+    public ApiResponse<AttendanceResponse> checkIn(@RequestParam Long userId,
```
@RequestBody CheckInRequest로 받으면 코드가 더 깔끔해질 것 같아요! 다른 메서들에서도요!

hyeblee replied:
`@PostMapping("/check-out")
public ApiResponse<AttendanceResponse> checkOut(@RequestBody CheckoutRequest request) {
    try {
        AttendanceResponse response = attendanceService.checkOut(request.getUserId(), request.getCheckOutTime());
        return ApiResponse.success("퇴근 기록이 성공적으로 등록되었습니다.", response);
    } catch (Exception e) {
        return ApiResponse.failure("퇴근 기록 등록에 실패했습니다: " + e.getMessage());
    }
}
`

junbem committed:
security: DB 접속 정보 환경 변수화

- 하드코딩된 DB 접속 정보를 환경 변수로 변경
- 기본값 설정으로 로컬 개발 환경 지원
- 민감한 정보의 GitHub 노출 방지

junbem committed:
refactor: 출근/퇴근 API를 @RequestBody 방식으로 변경

- @RequestParam → @RequestBody로 변경하여 RESTful API 설계 개선
- CheckInRequest, CheckOutRequest DTO 활용
- JSON Body 방식으로 API 호출 방식 변경
- 확장성 및 타입 안전성 향상
